### PR TITLE
Add screenshot files cleanup option to Housekeeping

### LINKED ISSUE #189: Allow to automatically delete old screenshots using housekeeping
### Problem

Most of the times, screenshots are only needed once and after taking them are uploaded somewhere.
They tend to clutter up the system.

### Proposal

Add a checkbox for "screenshots" in the housekeeping setup step

### Prior Art (Optional)

_No response_

### PR DESCRIPTION
**Pull request by Antolius**
Resolve #189 

Add check button for enabling screenshot files cleanup to Housekeeping view. Add a custom icon for screenshot files.

Mirrors changes to the switchboard plugin proposed in [this PR](https://github.com/elementary/switchboard-plug-security-privacy/pull/141), part of solving the [Add screenshots to housekeeping #55 task on settings-daemon](https://github.com/elementary/settings-daemon/issues/55).

Screenshot from 2023-05-08 00 52 18

### CODE REVIEW

File: data/folder-screenshots-icon/24.svg
Comment (Antolius) [inline data/folder-screenshots-icon/24.svg:1]: I copied this icon from the [switchboard pull request](https://github.com/elementary/switchboard-plug-security-privacy/pull/141). If having the same custom icon in 2 different repositories is not OK I could:

1. Pull it into [Icons](https://github.com/elementary/icons) repo and reuse it in both applications. Although I'm not sure how appropriate that would be considering that this is not a standard icon.
2. Load switchboard-plug-security-privacy icons here and just reuse it that way.

Comment (danirabbit) [reply data/folder-screenshots-icon/24.svg:1]: I think the precedent we have so far is in keyboard settings we gresource the Screenshots app icon, which you can find here: https://github.com/elementary/screenshot/blob/master/data/icons/24.svg

File: data/io.elementary.onboarding.gresource.xml [hunk 1]
```diff
@@ -6,5 +6,7 @@
     <file compressed="true" preprocess="xml-stripblanks">appearance-default.svg</file>
     <file compressed="true" preprocess="xml-stripblanks">appearance-dark.svg</file>
     <file compressed="true" preprocess="xml-stripblanks">appearance-scheduled.svg</file>
+    <file alias="24x24@2/apps/folder-screenshots-icon.svg" compressed="true" preprocess="xml-stripblanks">folder-screenshots-icon/24.svg</file>
+        <file alias="24x24/apps/folder-screenshots-icon.svg" compressed="true" preprocess="xml-stripblanks">folder-screenshots-icon/24.svg</file>
```
Comment (danirabbit) [inline data/io.elementary.onboarding.gresource.xml:10]: There's actually a namespacing trick we can do to load these into the icons path automatically:
```suggestion
</gresource>
<gresource prefix="io/elementary/onboarding/icons">
    <file alias="24x24@2/apps/folder-screenshots-icon.svg" compressed="true" preprocess="xml-stripblanks">folder-screenshots-icon/24.svg</file>
    <file alias="24x24/apps/folder-screenshots-icon.svg" compressed="true" preprocess="xml-stripblanks">folder-screenshots-icon/24.svg</file>
```

File: src/Application.vala [hunk 2]
```diff
@@ -24,6 +24,8 @@ public class Onboarding.App : Gtk.Application {
         granite_settings.notify["prefers-color-scheme"].connect (() => {
             gtk_settings.gtk_application_prefer_dark_theme = granite_settings.prefers_color_scheme == Granite.Settings.ColorScheme.DARK;
         });
+
+        Gtk.IconTheme.get_for_display (Gdk.Display.get_default ()).add_resource_path ("/io/elementary/onboarding");
```
Comment (danirabbit) [inline src/Application.vala:28]: We don't actually need this if we namespace the gresource to the icons path
```suggestion
```

**Review by danirabbit — CHANGES_REQUESTED** [review]
Nice work! Just a couple of changes

File: src/Application.vala (hunk 2 above)
Comment (Antolius) [reply src/Application.vala]: One potential problem: for this to work GTK application_id needs to match, and it previously did not.
I changes it in [this commit](https://github.com/elementary/onboarding/pull/208/commits/d9016162c6301d5183c8b8ec5362fbb45442618f#diff-f4e5d48d751c670a8575715da1cb0a0976532427a8b6143c7fdede415a33ba6cR8), but I'm not sure if this is 100% safe to do. In my testing it works OK, but there might be some backwards compatibility issue I'm not aware of.

**Review by danirabbit — APPROVED** [review]
Looks good, works as expected. Thanks for fixing the app ID! Nice work :)

**Pull request merged**